tidb/src/main/java/io/tidb/bigdata/tidb/codec/TiDBEncodeHelper.java

```diff
@@ -320,10 +325,13 @@ public List<BytePairWrapper> generateKeyValuesByRow(Row row) {
     keyValues.add(generateRecordKeyValue(row, handle, false));
     keyValues.addAll(generateIndexKeyValues(row, handle, false));
 
+    // if the key needs to be updated, then the deletion pair need to be removed and keep
+    // the update pair. Otherwise, the execution order can't be guaranteed, which may cause
+    // a problem when the update pair is executed first.
```

Comment:
how about explaining what's problem would be caused? such as ".. may cause the key-value delete when the... and delete pair is executed second".